core/src/gl/shaderProgram.cpp

```diff
@@ -281,34 +278,39 @@ std::string ShaderProgram::applySourceBlocks(const std::string& source, bool fra
         sourceOut << "#define TANGRAM_VERTEX_SHADER\n";
     }
 
-    auto sourcePos = source.begin();
+    std::stringstream sourceIn(source);
+    std::string line;
 
-    for (auto it = source.begin(); std::regex_search(it, source.end(), sm, pragmaLine);) {
-        it += sm.position() + sm.length();
+    while (std::getline(sourceIn, line)) {
+        if (line.empty()) {
+            continue;
+        }
 
-        std::string pragmaName = sm[1];
-        bool unique;
-        std::tie(std::ignore, unique) = pragmas.emplace(std::move(pragmaName));
+        sourceOut << line << '\n';
```

Comment:
Just tested, and this is definitely an improvement!

Do you think we can apply the "Replace blackslash+newline with spaces" part at this point on `line` variable.
If we find `\\` on `line` variable directly and not inject `\n` on the stream at this point?